evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -142,4 +142,13 @@ def validate(self, X, y):
                                                      message_code=DataCheckMessageCode.TARGET_INCOMPATIBLE_OBJECTIVE,
                                                      details=details).to_dict())
 
+        if X is not None:
+            X = _convert_to_woodwork_structure(X)
+            X_index = list(X.to_dataframe().index)
+            y_index = list(y_df.index)
+            if X_index != y_index:
```

Comment:
I think this makes sense! pandas treats `RangeIndex` and `Int64Index` the same as long as they have the same values when it comes to doing operations on different series/dataframes so no NaNs will be imputed. So I think list comparison is a good idea.